llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -33,7 +33,12 @@ using namespace llvm;
 #include "RISCVGenInstrInfo.inc"
 
 RISCVInstrInfo::RISCVInstrInfo(RISCVSubtarget &STI)
-    : RISCVGenInstrInfo(RISCV::ADJCALLSTACKDOWN, RISCV::ADJCALLSTACKUP),
+    : RISCVGenInstrInfo(RISCVABI::isCheriPureCapABI(STI.getTargetABI())
```

Comment:
There are currently 37 cases of `RISCVABI::isCheriPureCapABI\(.+getTargetABI`, and only 7 that use a different variable. I feel like adding a STI.isCheriPureCapABI() would make things slightly more readable since clang-format will do less weird stuff with shorter names.

Reply:
Yeah, probably. I initially didn't because there's no such thing for the hard-float ABIs (though they don't even have helpers either, they just switch in the few places it's needed), but purecap is rather more invasive.